src/chainstate/stacks/boot/pox_2_tests.rs

```diff
@@ -483,3 +483,324 @@ fn test_simple_pox_lockup_transition_pox_2() {
+        .block_height_to_reward_cycle(burnchain.pox_constants.v1_unlock_height as u64)
+        .unwrap()
+        + 1;
+
+    eprintln!("First v2 cycle = {}", first_v2_cycle);
+
+    let epochs = StacksEpoch::all(0, 25 + 10);
+
+    let observer = TestEventObserver::new();
+
+    let (mut peer, mut keys) = instantiate_pox_peer_with_epoch(
+        &burnchain,
+        "test_pox_extend_transition_pox_2",
+        6002,
+        Some(epochs.clone()),
+        Some(&observer),
+    );
+
+    let num_blocks = 35;
+
+    let alice = keys.pop().unwrap();
+    let bob = keys.pop().unwrap();
+    let charlie = keys.pop().unwrap();
+
+    let mut alice_reward_cycle = 0;
+
+    for tenure_id in 0u32..num_blocks {
+        let microblock_privkey = StacksPrivateKey::new();
+        let microblock_pubkeyhash =
+            Hash160::from_node_public_key(&StacksPublicKey::from_private(&microblock_privkey));
+        let tip = SortitionDB::get_canonical_burn_chain_tip(&peer.sortdb.as_ref().unwrap().conn())
+            .unwrap();
+
+        let (burn_ops, stacks_block, microblocks) = peer.make_tenure(
+            |ref mut miner,
+             ref mut sortdb,
+             ref mut chainstate,
+             vrf_proof,
+             ref parent_opt,
+             ref parent_microblock_header_opt| {
+                let parent_tip = get_parent_tip(parent_opt, chainstate, sortdb);
+                let coinbase_tx = make_coinbase(miner, tenure_id as usize);
+
+                let mut block_txs = vec![coinbase_tx];
+
+                if tenure_id == 1 {
+                    // Alice locks up exactly 25% of the liquid STX supply, so this should succeed.
+                    // this locks for cycles 6, 7, 8, 9.
+                    //  however, the v2 unlock occurs before cycle 8, so alice will be unlocked before cycles 8 and 9
+                    let alice_lockup = make_pox_lockup(
+                        &alice,
+                        0,
+                        1024 * POX_THRESHOLD_STEPS_USTX,
+                        AddressHashMode::SerializeP2PKH,
+                        key_to_stacks_addr(&alice).bytes,
+                        4,
+                        tip.block_height,
+                    );
+                    block_txs.push(alice_lockup);
+                } else if tenure_id == 10 {
+                    // Lets have Bob lock up for v2
+                    // this will lock for cycles 8, 9, 10, and 11
+                    //  the first v2 cycle will be 8
+                    let bob_lockup = make_pox_2_lockup(
+                        &bob,
+                        0,
+                        512 * POX_THRESHOLD_STEPS_USTX,
+                        AddressHashMode::SerializeP2PKH,
+                        key_to_stacks_addr(&bob).bytes,
+                        4,
+                        tip.block_height,
+                    );
+                    block_txs.push(bob_lockup);
+
+                    // Alice _will_ auto-unlock, so stack in PoX v2
```

Comment:
how are you using the word "so" here? she is stacking becasue she's unlocking, or are these two different events?